react-components/src/components/CacheProvider/AssetMappingCache.ts

```diff
@@ -80,6 +123,58 @@ export class AssetMappingCache {
     return await this.fetchAndCacheMappingsForModel(modelId, revisionId);
   }
 
+  private async fetchAndCacheMappingsForIds(
+    modelId: ModelId,
+    revisionId: RevisionId,
+    ids: number[],
+    filterType: string
+  ): Promise<AssetMapping[]> {
+    const key: ModelNodeIdKey = modelRevisionNodesToKey(modelId, revisionId, ids);
+    const idChunks = chunk(ids, 100);
+    const assetMappingsPromises = idChunks.map(async (idChunk) => {
+      const filter = filterType === 'nodeIds' ? { nodeIds: idChunk } : { assetIds: idChunk };
+      return await this._sdk.assetMappings3D
+        .filter(modelId, revisionId, {
+          limit: 1000,
+          filter
+        })
+        .autoPagingToArray({ limit: Infinity });
+    });
+    const assetMappingsArrays = await Promise.all(assetMappingsPromises);
+    const assetMappings = assetMappingsArrays.flat();
+    this._nodeAssetIdsToAssetMappings.set(key, Promise.resolve(assetMappings));
+    return assetMappings;
+  }
+
+  private async getAssetMappingsForNodes(
+    modelId: ModelId,
+    revisionId: RevisionId,
+    nodes: Node3D[]
+  ): Promise<AssetMapping[]> {
+    const nodeIds = nodes.map((node) => node.id);
+    const key: ModelNodeIdKey = modelRevisionNodesToKey(modelId, revisionId, nodeIds);
+    const cachedResult = this._nodeAssetIdsToAssetMappings.get(key);
+
+    if (cachedResult !== undefined) {
+      return await cachedResult;
+    }
+    return await this.fetchAndCacheMappingsForIds(modelId, revisionId, nodeIds, 'nodeIds');
+  }
+
+  private async getAssetMappingsForAssetIds(
+    modelId: ModelId,
+    revisionId: RevisionId,
+    assetIds: number[]
+  ): Promise<AssetMapping[]> {
+    const key: ModelNodeIdKey = modelRevisionNodesToKey(modelId, revisionId, assetIds);
```

Comment:
I see we use this function for both model/revision and asset IDs. Should we rename it?